csv/csv.go

```diff
@@ -0,0 +1,93 @@
+package csv
+
+import (
+	"encoding/csv"
+	"fmt"
+	"log"
+	"main/repository"
+	"os"
+	"reflect"
+
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+func ExportCsv(telID string, db *gorm.DB) (string, error) {
+	u := repository.NewGormUser(db)
+
+	// Get user by telegramID
+	user, err := u.GetByTelegramId(telID)
+	if err != nil {
+		log.Println("Couldn't find user: ", err)
+		return "", err
+	}
+
+	fileName := fmt.Sprintf("./telegram/%s.csv", uuid.New()) // Unique file name
+	// Create a CSV file
+	f, err := os.Create(fileName)
+
+	if err != nil {
+		log.Println("Couldn't create csv file: ", err)
+		return "", err
+	}
+	defer f.Close()
+
+	w := csv.NewWriter(f)
+	defer w.Flush()
+
+	// Write headers
+	header := []string{"لینک", "طول جغرافیایی", "عرض جغرافیایی", "توضیحات", "قیمت فروش", "قیمت اجاره", "مقدار رهن", "شهر", "محله", "متراژ", "تعداد اتاق", "دسته بندی", "سن بنا", "دسته بندی آپارتمان/ویلا", "طبقه", "انباری", "آسانسور", "عنوان آگهی", "لینک عکس"}
+	if err := w.Write(header); err != nil {
+		log.Println("error writing header to csv:", err)
+		return "", err
+	}
+
+	for _, v := range user.Ads {
+		val := reflect.ValueOf(v).Elem() // Get the value of the pointer
+		record := make([]string, 0)
+
+		for i := 0; i < (val.NumField() - 1); i++ {
+			field := val.Field(i)
+
+			//skip: CreatedAt, UpdatedAt, DeletedAt,  ID, UniqueId, NumberOfViews, Users(last field)
+			if i != 0 && i != 1 && i != 2 && i != 3 && i != 5 && i != 9 {
+				var fieldValue string
+				// Check for CategoryPR to insert specific string values
+				if i == 17 { // Assuming CategoryPR is at index 17
```

Comment:
Whenever you need to leave a comment, it indicates that your code may be poorly written.